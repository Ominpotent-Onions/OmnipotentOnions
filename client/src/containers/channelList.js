import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchChannels, fetchOneGroup } from '../actions';
import _ from 'lodash';

import NewChannel from './new_channel';

import { Icon, Menu, Input } from 'semantic-ui-react';

export class ChannelList extends Component {
  constructor(props) {
    super(props);
  }

  // componentWillMount() {
  //   this.props.fetchOneGroup(this.props.groupId);
  // }

  // renderChannels() {
  //   return _.map(this.props.channels, channel => {
  //     this.props.socket.emit('subscribe', channel.id);
  //     return (
  //       <Menu.Item key={channel.id}
  //                  name={channel.name}
  //                  active={this.state.activeItem === channel.name} 
  //                  onClick={(e, d) => {
  //                    console.log('yOOOOOOOO'),
  //                    this.handleItemClick(e, d),
  //                    this.props.handleMessage; 
  //                  }}>
  //         {channel.name}
  //       </Menu.Item>
  //     );
  //   });
  // }

  // render() {
  //   return (
  //     <Menu.Item
  //       key={this.props.channel.id}
  //       name={this.props.channel.name}
  //       active={this.props.activeItem === this.props.channel.name}
  //       value={this.props.channel.id}
  //       onClick={(e, d) => { this.props.handleItemClick(e, d), this.props.handleMessage(d.value); }}
  //     >
  //       {this.props.channel.name}
  //     </Menu.Item>
  //   );
  // }
}

const mapStateToProps = function(state) {
  return { channels: state.channels, oneGroup: state.oneGroup };
};

export default connect(mapStateToProps, { fetchChannels, fetchOneGroup })(ChannelList);
