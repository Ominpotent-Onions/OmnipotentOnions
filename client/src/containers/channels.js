import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchChannels, fetchOneGroup, fetchGroups } from '../actions';
import _ from 'lodash';

import NewChannel from './new_channel';
import ChannelList from './channelList';

import { Icon, Menu } from 'semantic-ui-react';


export class Channels extends Component {
  constructor(props) {
    super(props);
    this.state = {
      activeItem: '',
    };
    this.handleItemClick = this.handleItemClick.bind(this);
  }

  componentWillMount() {
    this.setState({
      activeItem: this.props.channelName
    });
  }

  handleItemClick(e, {name}) {
    console.log(e);
    this.setState({activeItem: name});
  }

  //creates a list of all channels

  renderChannels() {
    console.log('CHANNELS: ', this.props.channels);
    return _.map(this.props.channels, channel => {
      this.props.socket.emit('subscribe', channel.id);
      return (
        <Menu.Item 
          key={channel.id}
          active={this.state.activeItem === channel.name}
          value={channel.id}
          name={channel.name}
          onClick={(e, d) => {
            this.handleItemClick(e, d),
            this.props.handleMessage(e, d); 
          }}
        >
          {channel.name}
        </Menu.Item>
      );
    });
  }

  render() {
    return (
      <Menu.Item>
        <Menu.Header>
          Channels
        </Menu.Header>
        <Menu.Menu>
          {this.renderChannels()}
        </Menu.Menu>
        <NewChannel groupId={this.props.groupId}/>
      </Menu.Item>
    );
  }
}

const mapStateToProps = function(state) {
  return { channels: state.channels, oneGroup: state.oneGroup };
};

export default connect(mapStateToProps, { fetchChannels, fetchOneGroup, fetchGroups })(Channels);
