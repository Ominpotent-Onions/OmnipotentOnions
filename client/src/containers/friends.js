import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchProfile, fetchFriends, fetchPendingRequests, fetchFriendRequests } from '../actions';

import FriendsList from './friends_list';
import PendingList from './friends_pending';


export class Friends extends Component {
  constructor(props) {
    super(props);
    this.props.fetchProfile(myUser);
    this.props.fetchFriends(myUser.id); // add user id to it at some point
    this.props.fetchPendingRequests();
    this.props.fetchFriendRequests();
  }

  render() {
    return (
      <div>
        <div> <FriendsList /> </div>
        <div> <PendingList /> </div>
      </div>
    );
  }
}

const mapStateToProps = function(state) {
  return { profile: state.profile };
}

export default connect(mapStateToProps, { fetchProfile, fetchFriends, fetchPendingRequests, fetchFriendRequests })(Friends);