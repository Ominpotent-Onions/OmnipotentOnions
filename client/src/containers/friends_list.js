import React, { Component } from 'react';
import { connect } from 'react-redux';
import _ from 'lodash';

export class FriendsList extends Component {

  renderFriends() {
    return _.map(this.props.friends, (friend) => {
      return (
        <div>
          <div> Name: {friend.display} </div>
          <div> Email: {friend.email} </div>
        </div>
      );
    });
  }

  render() {
    return (
      <div>
        <h3>Friends List</h3>
        {this.renderFriends()}
      </div>
    );
  }
}

let mapStateToProps = function(state) {
  return { friends: state.friends };
};

export default connect(mapStateToProps, null)(FriendsList);