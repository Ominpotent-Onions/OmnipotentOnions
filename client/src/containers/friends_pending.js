import React, { Component } from 'react';

export class PendingList extends Component {
  render() {
    return (
      <div>Pending List</div>
    );
  }
}

export default PendingList;