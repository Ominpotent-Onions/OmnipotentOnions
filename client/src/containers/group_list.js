import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchGroups } from '../actions';

import InviteLink from './invite_link';
import NewGroup from './new_group';
import JoinGroup from './join_group';
import LeaveGroup from './leave_group';

import { Icon, Menu, Input } from 'semantic-ui-react';

class GroupList extends Component { 
  constructor(props) {
    super(props);
    this.state = {
      showInvite: false,
      
    };
    this.shortIdView = this.shortIdView.bind(this);
  }

  shortIdView() {
    const message = 'Share this code with others to join the group! :';  
    // alert(message + '\n' + this.props.group.shortID);
    this.setState({
      showInvite: !this.state.showInvite
    });
  }

  render() {
    return (
      <Menu.Item className='groupName'        
        key={this.props.groups.group_id}
        active={this.props.activeItem === this.props.group.groups.name}>
        <div
          style={{display: 'inline'}}
          key={this.props.group.group_id}
          name={this.props.group.groups.name}
          value={this.props.group.group_id} 
          onClick={(e, d) => { 
            this.props.handleItemClick(this.props.group.groups.name), 
            this.props.handleChannel(this.props.group.group_id); 
          }}
        >
          {this.props.group.groups.name}
        </div> 
        <InviteLink 
          group={this.props.group.groups} 
          profile={this.props.profile} showShortId={this.shortIdView}/>
        <LeaveGroup 
          group={this.props.group} 
          handleDeleteGroup={this.props.handleDeleteGroup}/>
        {
          this.state.showInvite ? 
            <Input
              className='sharelink'
              color='teal'
              inverted color='teal' 
              value={`share this code: ${this.props.group.groups.shortID}`}
              transparent={true}
            /> : 
            null
        }
      </Menu.Item>
    );
  }
}

const mapStateToProps = function(state) {
  return { groups: state.groups, profile: state.profile };
};

export default connect(mapStateToProps, { fetchGroups })(GroupList);
