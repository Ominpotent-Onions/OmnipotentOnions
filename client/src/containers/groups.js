import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchGroups } from '../actions';

import InviteLink from './invite_link';
import NewGroup from './new_group';
import JoinGroup from './join_group';
import LeaveGroup from './leave_group';

import { Segment, Icon } from 'semantic-ui-react';

class Groups extends Component { 
  constructor(props) {
    super(props);
  }

  componentWillMount() {
    this.props.fetchGroups(this.props.profile);    
  }

  renderGroups() {
    return _.map(this.props.groups, group => {
      return (
        <Segment key={group.id}>
          <button onClick={this.props.handleChannel} value={group.id}> {group.groups.name} </button>
          <InviteLink group={group.groups} profile={this.props.profile}/>
          <LeaveGroup group={group}/>
        </Segment>
      );
    });
  }

  render() {
    return (
      <div>
        <h2>Groups</h2>
        <Segment.Group>
          {this.renderGroups()}
          <NewGroup profile={this.props.profile}/>
          <JoinGroup profile={this.props.profile}/>
          <button onClick={this.props.showEvents}>
            <Icon name='calendar' size='big' />          
          </button>
        </Segment.Group>
      </div>
    );
  }
}

const mapStateToProps = function(state) {
  return { groups: state.groups, profile: state.profile };
};

export default connect(mapStateToProps, { fetchGroups })(Groups);