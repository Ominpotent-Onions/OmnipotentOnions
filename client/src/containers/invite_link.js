import React, { Component } from 'react';
import { connect } from 'react-redux';
import { createInvite } from '../actions';

import shortid from 'shortid';

class InviteLink extends Component {
  componentDidMount() {
    if(!this.props.group.shortID){
      this.props.group.shortID = shortid.generate();
    }
  }


  handleGroupInviteButton() {
    const message = 'Share this code with others to join the group! :';
    const shortID = this.props.group.shortID;

    console.log(message + '\n' + shortID);
  }

  render() {    
    return (
      <div>
        <button onClick={() => this.handleGroupInviteButton()}>+</button>
      </div>
    );
  }
}

export default InviteLink;