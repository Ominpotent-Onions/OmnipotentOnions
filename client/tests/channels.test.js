import React from 'react';
import { Channels } from '../src/containers/channels'
import renderer from 'react-test-renderer';
import _ from 'lodash';

const fetchChannels = function(group) {
  return;
}

const channels = [
  {
    id: 1,
    name: 'Shi-Hao\'s Smashing Channel'
  },
  {
    id: 2,
    name: 'Dylan\'s Dynamic Channel'
  },
  {
    id: 3,
    name: 'Peter\'s Poetic Channel'
  },
  {
    id: 4,
    name: 'Gideon\'s Gallant Channel'
  }
];


it('renders correctly', () => {
  const tree = renderer.create(<Channels 
    fetchChannels={fetchChannels}
    channels={channels}
  />);

  expect(tree).toMatchSnapshot();
})