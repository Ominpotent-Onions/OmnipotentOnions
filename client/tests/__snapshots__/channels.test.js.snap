// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h2>
    Channels
  </h2>
  <div
    className="ui segments"
  >
    <div
      className="ui segment"
    >
      <div>
        Shi-Hao's Smashing Channel
      </div>
    </div>
    <div
      className="ui segment"
    >
      <div>
        Dylan's Dynamic Channel
      </div>
    </div>
    <div
      className="ui segment"
    >
      <div>
        Peter's Poetic Channel
      </div>
    </div>
    <div
      className="ui segment"
    >
      <div>
        Gideon's Gallant Channel
      </div>
    </div>
  </div>
</div>
`;
